### DISCUSSION TITLE
Can't upload videos longer then 1min or larger then 100+MB

### DISCUSSION THREAD

**Discussion by al73rus** [immich-app/immich#13573] [Q&A]
### The bug

I have some videos on my phone which i can't upload. They are 1+min long or 100+MB in size.
Mobile app doen't give an error just doesn't upload it and never try again, on webpage it just ends up with an error (no info on what error, just can't upload).
I tried downloading test vid longer/larger then i mentioned on desktop and still didn't work.
All vid bellow that level are uploading without a problem.
Didn't change any settings or upgraded storage etc. It started around 2-3 weeks ago (maybe from version 116 can't point it directly to what wersion)

### The OS that Immich Server is running on

Unraid 7 beta 3 

### Version of Immich Server

Latest

### Version of Immich Mobile App

Latest

### Platform with the issue

- [ ] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Unraid docker
```

### Your .env content

```Shell
None
```

### Reproduction steps

1. Open app/page
2. Select a video longer than 1+min ir larger than 100+MB
3. Upload and get ann error on webpage, on the app there is no info.
...

### Relevant log output

```shell
There is no error after failed atempt.got only info with the asset uploading and then nothing
```

**Comment by Mraedis** [comment]
Are you using cloudflare proxy? It has a 100MB limit.

**Comment by al73rus** [comment]
Yes i am, but still tried to connect to the app localy in the app with the local adress at home and still i didn't upload it.

**Comment by al73rus** [comment]
Will double chec when i get home and try it once again on mobilr and pc

**Comment by ALERTua** [comment]
Solution:
- Open https://dash.cloudflare.com
- Change your Immich DNS Record to **DNS Only** **CNAME** so it leads to your duckdns Immich counterpart
![image](https://github.com/user-attachments/assets/ae615f88-dcf4-4e6a-b9e7-153813853f9c)
- Listen for this address in your Nginx Proxy Manager and redirect it to the Immich instance
![image](https://github.com/user-attachments/assets/79f11429-9265-4cc5-b61f-4d2a03a41b6c)
![image](https://github.com/user-attachments/assets/b2fd8410-b0fc-4ec0-9968-87746d7e2c03)

**Reply by teeouan** [reply]
thx kub

**Reply by techsk8** [reply]
Thank you! It works great.
Although it's not he best solution, cause it exposes my real IP address out to the world, which is why I'm using the proxied Cloudflare tunnels in the first place.

**Comment by truncsphere** [comment]
I'm seeing the same issue using Cloudflare proxied DNS  from a video taken yesterday.

**Comment by al73rus** [comment]
Thank you for helping out with ngnix. Just set it up after watching few vida and now it's running like a charm.
Thanks again for your help.

**Comment by ChocolateLoverRaj** [comment]
It would be convenient if Immich split the upload into chunks that are within Cloudflare's limits, or was able to pick up where it left off it it encountered an error uploading. 

I am using Cloudflared tunnels. I may be having this same issue, I will try through port forwarding without a proxy and see if it works.

**Comment by powerbling** [comment]
I also think chunking could be a nice upgrade

**Comment by cybersader** [comment]
The only way to account for all of these issues is to indeed find a chunking library that is future-resilient to changes in Immich development.

[[Feature]: Upload large files in chunks · immich-app/immich · Discussion #1674](https://github.com/immich-app/immich/discussions/1674#discussioncomment-11065843)

**Reply by ChocolateLoverRaj** [reply]
Even better: Instead of doing chunked uploads always, just make it so that failed uploads keep their progress and retries continue uploading where they left off.

**Comment by deanfourie1** [comment]
Do we know if there is anything in the pipeline for the team to implement chunked uploading? Is this something they are going to address? Asking as I need to know if I need to switch to tailscale tunnel.

**Reply by Mraedis** [reply]
It's on the To Do list but no concrete ETA.

**Reply by deanfourie1** [reply]
Thanks

**Reply by deanfourie1** [reply]
Also I just noticed the app now has a multi connection mode similar to home assistant. So that will be find for now :)

**Comment by fanshaohua-fan** [comment]
I bypass the issue via the multi-networking setup as well.
So within my local network, the app connects to the server(hosted on my PC) through the internal IP(192.168.x.xxx).
In other cases, the app connects to the server via the public domain proxied via Cloudflare.

**Reply by deanfourie1** [reply]
Not ideal though, if you take a super important video then loose your phone your faaarked.

**Reply by techieanant** [reply]
@fanshaohua-fan can you describe how your setup for multi-networking?

**Reply by techsk8** [reply]
It's pretty straight forward, just make sure you have the latest version of the app and go into settings -> networking -> and it's pretty simple from there, the only inconvenient is that you need to keep your location turned all at all times, so that the app can check if you're connected to the designated WIFI network, where the server can be found.

**Reply by fanshaohua-fan** [reply]
> @fanshaohua-fan can you describe how your setup for multi-networking?

@techieanant For the local network, enter your Wi-Fi name and the Immich URL with the internal IP address. 
It’s better to set up the external network as well, so the server remains accessible outside your local Wi-Fi.

Here is my setting:
![local](https://github.com/user-attachments/assets/be7700a4-a4a4-4046-b5cc-f25ee3b7ca39)
![external](https://github.com/user-attachments/assets/668c9ae6-ccad-491a-8266-63a4f2def7fa)

**Reply by sginestrini** [reply]
> > @fanshaohua-fan can you describe how your setup for multi-networking?
> 
> @techieanant For the local network, enter your Wi-Fi name and the Immich URL with the internal IP address. It’s better to set up the external network as well, so the server remains accessible outside your local Wi-Fi.
> 
> Here is my setting: ![local](https://private-user-images.githubusercontent.com/6319967/429058808-be7700a4-a4a4-4046-b5cc-f25ee3b7ca39.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY1MDY4NzksIm5iZiI6MTc2NjUwNjU3OSwicGF0aCI6Ii82MzE5OTY3LzQyOTA1ODgwOC1iZTc3MDBhNC1hNGE0LTQwNDYtYjVjYy1mMjVlZTNiN2NhMzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMjNUMTYxNjE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTQ3OTg0YjdkNGZlZjM1NDE2NmVhN2NlYzgwYjdkYWU3MmQzZTA5NGY4NjcyMzNiNjkxNTgxYWFkNTQxYTJiNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.PajAj8UNz8r2zvM4gciIm-Z09D_LUkS2t_0mo7itQso) ![external](https://private-user-images.githubusercontent.com/6319967/429058801-668c9ae6-ccad-491a-8266-63a4f2def7fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjY1MDY4NzksIm5iZiI6MTc2NjUwNjU3OSwicGF0aCI6Ii82MzE5OTY3LzQyOTA1ODgwMS02NjhjOWFlNi1jY2FkLTQ5MWEtODI2Ni02M2E0ZjJkZWY3ZmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMjNUMTYxNjE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTIwNTRhY2M1MDk3MzhiN2QwZGQ4ZDBkZDU1OTQ1MDgyMDFlMTE2YzE0YzcwYjZmMTRiMmI5MjZmY2NmODllOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.3kT0rYlDusGRdkpkDR-vy8GPcHA2JzNjMjMtpUJMzR0)

In order for that to work also outside you need to setup as external network both internal IP and external URL and put the internal IP as first choice. By that you can just turn on a vpn when you are outside and immich will try the internal ip even you are not on the local wifi network.

**Reply by LudnicaKiller** [reply]
> > @fanshaohua-fan can you describe how your setup for multi-networking?
> 
> @techieanant For the local network, enter your Wi-Fi name and the Immich URL with the internal IP address. It’s better to set up the external network as well, so the server remains accessible outside your local Wi-Fi.
> 
> Here is my setting: ![local](https://private-user-images.githubusercontent.com/6319967/429058808-be7700a4-a4a4-4046-b5cc-f25ee3b7ca39.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2MzUwNDEsIm5iZiI6MTc3MDYzNDc0MSwicGF0aCI6Ii82MzE5OTY3LzQyOTA1ODgwOC1iZTc3MDBhNC1hNGE0LTQwNDYtYjVjYy1mMjVlZTNiN2NhMzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMDlUMTA1OTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjdlMjhkNTgxNTU2MjI1NDE4MjM3Mzc2ZWExOGI1M2UzM2NkMDliYWE5MzMyZWI3Y2U4NTQzZTk0MDcxNTQ0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.JjgvPdf158tfRrjE8RKUHceIm5CZcLIfDiNqCLY_SLw) ![external](https://private-user-images.githubusercontent.com/6319967/429058801-668c9ae6-ccad-491a-8266-63a4f2def7fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2MzUwNDEsIm5iZiI6MTc3MDYzNDc0MSwicGF0aCI6Ii82MzE5OTY3LzQyOTA1ODgwMS02NjhjOWFlNi1jY2FkLTQ5MWEtODI2Ni02M2E0ZjJkZWY3ZmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMDlUMTA1OTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjY3OTJkZGVjOWNkZjYxMjEyNTNhYzgxYWZmZTk2OWI0OTA5MWVkMzMwNTE0MzdhZDNkZjY5NjZhNjFhODE5OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.l2ml0MsIfAsBVz2hXY066_i0fR9vI6Iaj1PwjNfnc-k)

Thanks man, works
